easybuild/easyblocks/g/gamess_us.py

```diff
@@ -0,0 +1,232 @@
+            r"GAMESS build directory\? \[.*\] ": self.installdir,  # building in install directory
+        }
+        run_cmd_qa(cmd, qa=qa, std_qa=stdqa, log_all=True, simple=True)
+
+        self.log.debug("Contents of install.info:\n%s" % read_file(os.path.join(self.builddir, 'install.info')))
+
+        # patch hardcoded settings in rungms to use values specified in easyconfig file
+        rungms = os.path.join(self.builddir, 'rungms')
+        try:
+            for line in fileinput.input(rungms, inplace=1, backup='.orig'):
+                line = re.sub(r"^(\s*set\s*TARGET)=.*", r"\1=%s" % self.cfg['ddi_comm'], line)
+                line = re.sub(r"^(\s*set\s*GMSPATH)=.*", r"\1=%s" % self.installdir, line)
+                line = re.sub(r"(null\) set VERNO)=.*", r"\1=%s" % self.version, line)
+                line = re.sub(r"^(\s*set DDI_MPI_CHOICE)=.*", r"\1=%s" % mpilib, line)
+                line = re.sub(r"^(\s*set DDI_MPI_ROOT)=.*%s.*" % mpilib.lower(), r"\1=%s" % mpilib_path, line)
+                line = re.sub(r"^(\s*set GA_MPI_ROOT)=.*%s.*" % mpilib.lower(), r"\1=%s" % mpilib_path, line)
+                # comment out all adjustments to $LD_LIBRARY_PATH that involves hardcoded paths
+                line = re.sub(r"^(\s*)(setenv\s*LD_LIBRARY_PATH\s*/.*)", r"\1#\2", line)
+                if self.cfg['scratch_dir']:
+                    line = re.sub(r"^(\s*set\s*SCR)=.*", r"\1=%s" % self.cfg['scratch_dir'], line)
+                    line = re.sub(r"^(\s*set\s*USERSCR)=.*", r"\1=%s" % self.cfg['scratch_dir'], line)
+                sys.stdout.write(line)
+        except IOError, err:
+            self.log.error("Failed to patch %s: %s" % (rungms, err))
+
+    def build_step(self):
+        """Custom build procedure for GAMESS-US: using compddi, compall and lked scripts."""
+        compddi = os.path.join(self.cfg['start_dir'], 'ddi', 'compddi')
+        run_cmd(compddi, log_all=True, simple=True)
+
+        # make sure the libddi.a library is present
+        libddi = os.path.join(self.cfg['start_dir'], 'ddi', 'libddi.a')
+        if not os.path.isfile(libddi):
+            self.log.error("The libddi.a library (%s) was never built" % libddi)
+        else:
+            self.log.info("The libddi.a library (%s) was successfully built." % libddi)
+
+        compall_cmd = os.path.join(self.cfg['start_dir'], 'compall')
+        compall = "%s %s %s" % (self.cfg['prebuildopts'], compall_cmd, self.cfg['buildopts'])
+        run_cmd(compall, log_all=True, simple=True)
+
+        cmd = "%s gamess %s" % (os.path.join(self.cfg['start_dir'], 'lked'), self.version)
+        run_cmd(cmd, log_all=True, simple=True)
+
+    def test_step(self):
+        """Run GAMESS-US tests (if 'runtest' easyconfig parameter is set to True)."""
+        if self.cfg['runtest']:
+            runall_cmd = "%s %s" % (os.path.join(self.builddir, 'runall'), self.version)
+            run_cmd(runall_cmd, path=self.builddir, log_all=True)
```

Comment:
TODO: run tests in a `tests` subdir, and check results from tests (check `exam<id>.log` for `EXECUTION OF GAMESS TERMINATED NORMALLY`)